app/src/main/java/school/cactus/hnmobile/NewStoriesAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package school.cactus.hnmobile
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.firebase.ui.database.FirebaseRecyclerAdapter
+import com.firebase.ui.database.FirebaseRecyclerOptions
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.database.ktx.database
+import com.google.firebase.database.ktx.getValue
+import com.google.firebase.ktx.Firebase
+import school.cactus.hnmobile.NewStoriesAdapter.StoryItemHolder
+
+class NewStoriesAdapter(options: FirebaseRecyclerOptions<Int>) :
+    FirebaseRecyclerAdapter<Int, StoryItemHolder>(options) {
+
+    private val query = Firebase.database.reference
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoryItemHolder {
+        // Create a new view, which defines the UI of the list item
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item_news, parent, false)
+        return StoryItemHolder(view)
+    }
+
+    override fun onBindViewHolder(storyItemHolder: StoryItemHolder, position: Int, model: Int) {
+
+        val queryStory = query.child("/v0/item/$model").child("title")
+
+        queryStory.addListenerForSingleValueEvent(object : ValueEventListener {
+
+            override fun onCancelled(error: DatabaseError) {
+            }
+
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val children = snapshot.getValue<String>()
```

Comment:
The name children is somehow misleading because it's plural but `snapshot.getValue<String>()` returns only a single String.

And also I am a bit confused now and didn't understand how you get the title only 🤔

Reply:
- I didn't notice. I will change the name, named "newstitle",

-when I request data from 

val queryStory = query.child("/v0/item/$model")
val newsTitle = snapshot!!.children

response is:

![WhatsApp Image 2020-12-24 at 6 04 29 PM](https://user-images.githubusercontent.com/53116879/103173200-449ff900-486a-11eb-9ad7-f187bf589441.jpeg)

Because of that I enter child name "title" to get value from.

val queryStory = query.child("/v0/item/$model").child("title")
val newsTitle = snapshot.getValue<String>()